superduperdb/ext/cohere/model.py

```diff
@@ -53,47 +56,25 @@ def pre_create(self, db):
             self.datatype = vector(self.shape)
 
     @retry
-    def _predict_one(self, X: str, **kwargs):
+    def predict_one(self, X: str):
         client = cohere.Client(get_key(KEY_NAME), **self.client_kwargs)
-        e = client.embed(texts=[X], model=self.identifier, **kwargs)
+        e = client.embed(texts=[X], model=self.identifier, **self.predict_kwargs)
         return e.embeddings[0]
 
     @retry
-    async def _apredict_one(self, X: str, **kwargs):
-        client = cohere.AsyncClient(get_key(KEY_NAME), **self.client_kwargs)
-        e = await client.embed(texts=[X], model=self.identifier, **kwargs)
-        await client.close()
-        return e.embeddings[0]
-
-    @retry
-    def _predict_a_batch(self, texts: t.List[str], **kwargs):
+    def _predict_a_batch(self, texts: t.List[str]):
         client = cohere.Client(get_key(KEY_NAME), **self.client_kwargs)
-        out = client.embed(texts=texts, model=self.identifier, **kwargs)
-        return [r for r in out.embeddings]
-
-    @retry
-    async def _apredict_a_batch(self, texts: t.List[str], **kwargs):
-        client = cohere.AsyncClient(get_key(KEY_NAME), **self.client_kwargs)
-        out = await client.embed(texts=texts, model=self.identifier, **kwargs)
-        await client.close()
+        out = client.embed(texts=texts, model=self.identifier, **self.predict_kwargs)
         return [r for r in out.embeddings]
 
-    def _predict(self, X, one=False, **kwargs):
-        if isinstance(X, str):
-            return self._predict_one(X)
+    def predict(self, dataset: t.Union[t.List, QueryDataset]) -> t.List:
         out = []
-        batch_size = kwargs.pop('batch_size', 100)
-        for i in tqdm.tqdm(range(0, len(X), batch_size)):
-            out.extend(self._predict_a_batch(X[i : i + batch_size], **kwargs))
-        return out
-
-    async def _apredict(self, X, one=False, **kwargs):
-        if isinstance(X, str):
-            return await self._apredict_one(X)
-        out = []
-        batch_size = kwargs.pop('batch_size', 100)
-        for i in range(0, len(X), batch_size):
-            out.extend(await self._apredict_a_batch(X[i : i + batch_size], **kwargs))
+        for i in tqdm.tqdm(range(0, len(dataset), self.batch_size)):
+            out.extend(
+                self._predict_a_batch(
+                    dataset[i : i + self.batch_size], **self.predict_kwargs
+                )
+            )
```

Comment:
It is necessary to implement a list of batch prediction logic in the parent class